Remotion/Web/Core/Resources/CachingApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,57 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+#if !NET48
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Rewrite;
+using Remotion.Utilities;
+
+namespace Remotion.Web.Resources;
+
+/// <summary>
+/// Extension methods for <see cref="RemotionStaticFilesCacheKeyRemovalRewriteRule"/>.
+/// </summary>
+public static class CachingApplicationBuilderExtensions
+{
+  /// <summary>
+  /// Enables static file serving with caching support for the current request path.
+  /// Replaces the call to <see cref="StaticFileExtensions.UseStaticFiles(Microsoft.AspNetCore.Builder.IApplicationBuilder)"/>.
+  /// </summary>
+  /// <remarks>
+  /// Requires <see cref="ResourcePathBuilderBasedStaticResourceCacheKeyProvider"/> to be useful.
+  /// </remarks>
+  public static IApplicationBuilder UseRemotionStaticFiles (this IApplicationBuilder builder, StaticFileOptions options, StaticFileCachingOptions cachingOptions)
+  {
+    ArgumentUtility.CheckNotNull(nameof(builder), builder);
+    ArgumentUtility.CheckNotNull(nameof(options), options);
+
+    builder.UseRewriter(new RewriteOptions().Add(new RemotionStaticFilesCacheKeyRemovalRewriteRule(options.RequestPath, true)));
```

Comment:
It is possible to use the rewrite middleware multiple times. Web Tests are currently not possible because we don't run the test site with .NET, which is required for the middleware stuff to run.

Reply:
> Web Tests are currently not possible because we don't run the test site with .NET, which is required for the middleware stuff to run.

Hence please create a ticket for re-motion 7 :)

Reply:
https://re-motion.atlassian.net/browse/RM-9132